front_end/src/components/question/private_note.tsx

```diff
@@ -0,0 +1,134 @@
+"use client";
+
+import { MDXEditorMethods } from "@mdxeditor/editor";
+import { isNil } from "lodash";
+import { useLocale, useTranslations } from "next-intl";
+import { FC, useEffect, useMemo, useRef, useState } from "react";
+
+import { savePrivateNote } from "@/app/(main)/questions/actions";
+import MarkdownEditor from "@/components/markdown_editor";
+import LoadingSpinner from "@/components/ui/loading_spiner";
+import RelativeTime from "@/components/ui/relative_time";
+import SectionToggle from "@/components/ui/section_toggle";
+import { useDebouncedCallback } from "@/hooks/use_debounce";
+import { Post } from "@/types/post";
+import { formatDate } from "@/utils/formatters/date";
+
+type Props = {
+  post: Post;
+};
+
+const SavedAgo: FC<{
+  savedAt: Date;
+}> = ({ savedAt }) => {
+  const [now, setNow] = useState(Date.now());
+  const diff = now - savedAt.getTime();
+  const t = useTranslations();
+  const locale = useLocale();
+
+  useEffect(() => {
+    if (diff < 60_000) {
+      // Schedule an update exactly when it crosses 60s
+      const timeout = setTimeout(() => {
+        setNow(Date.now());
+      }, 60_000 - diff);
+
+      return () => clearTimeout(timeout);
+    }
+  }, [diff]);
+
+  if (diff < 60_000) {
+    return t.rich("savedAgo", {
+      date: () => "just now",
+    });
+  }
+
+  return t.rich("savedAgo", {
+    date: () => (
+      <RelativeTime datetime={savedAt.toString()} format="relative">
+        {formatDate(locale, savedAt)}
+      </RelativeTime>
+    ),
+  });
+};
```

Comment:
I believe it is better to track only `isJustNow` (not `now`), schedule a single flip when we cross 60s from `savedAtMs`, and pass `toISOString()` to RelativeTime for a stable, parseable datetime

```suggestion
const SavedAgo: FC<{ savedAt: Date }> = ({ savedAt }) => {
  const t = useTranslations();
  const locale = useLocale();

  const savedAtMs = savedAt.getTime();
  const [isJustNow, setIsJustNow] = useState(
    () => Date.now() - savedAtMs < 60_000
  );

  useEffect(() => {
    const elapsed = Date.now() - savedAtMs;
    const remaining = 60_000 - elapsed;

    if (remaining <= 0) {
      setIsJustNow(false);
      return;
    }

    setIsJustNow(true);
    const id = window.setTimeout(() => setIsJustNow(false), remaining);
    return () => window.clearTimeout(id);
  }, [savedAtMs]);

  return t.rich("savedAgo", {
    date: () =>
      isJustNow ? (
        t('justNow')
      ) : (
        <RelativeTime datetime={savedAt.toISOString()} format="relative">
          {formatDate(locale, savedAt)}
        </RelativeTime>
      ),
  });
};
```